docs/fuse-box/tutorials/mobile-swap-tokens-guide.mdx

```diff
@@ -0,0 +1,404 @@
+                     ),
+               SizedBox(height: 20),
+               Text(_swapStatus),
+             ],
+           ),
+         ),
+       );
+     }
+   }
+   ```
+  
+In the `build` method, we return a `Scaffold` widget, which provides the basic structure for the screen. The `AppBar` contains the title 'Token Swap'. The `body` consists of a `Center` widget containing a `Column` with a `CircularProgressIndicator` or an `ElevatedButton` based on the `_isLoading` state. Below the button, we display the `_swapStatus` using a `Text` widget.
+
+Save the file. Run the application using the command
+
+```bash
+flutter run
+```
+
+At this point, you can confirm that the SWAP functionality works! The next step is a bonus step to improve the User Experience, and we will update the code to include a Dropdown for selecting currencies and an input field to specify a particular amount. Add the Token address variables:
+
+```dart
+ final amount = BigInt.parse(_amountController.text.trim());
+
+    final nativeTokenAddress = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE';
+    final usdcTokenAddress = '0x28C3d1cD466Ba22f6cae51b1a4692a831696391A';
+    final usdtTokenAddress = '0xFaDbBF8Ce7D5b7041bE672561bbA99f79c532e10';
+    final voltTokenAddress = '0x34Ef2Cc892a88415e9f02b91BfA9c91fC0bE6bD4';
+
+    String outputTokenAddress;
+    switch (_selectedToken) {
+      case 'usdc':
+        outputTokenAddress = usdcTokenAddress;
+        break;
+      case 'usdt':
+        outputTokenAddress = usdtTokenAddress;
+        break;
+      case 'volt':
+        outputTokenAddress = voltTokenAddress;
+        break;
+      default:
+        outputTokenAddress = usdcTokenAddress; // Default to usdc
+    }
+```
+
+Update the `InputAmount` to take the value from `amount`
+
+```dart
+  inputAmount: amount,
+```
+
+Update the `body` component:
+
+```dart
+ body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Form(
+          key: _formKey,
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: [
+              DropdownButtonFormField<String>(
+                value: _selectedToken,
+                onChanged: (value) {
+                  setState(() {
+                    _selectedToken = value!;
+                  });
+                },
+                items: [
+                  DropdownMenuItem(
+                    value: 'usdc',
+                    child: Text('USDC'),
+                  ),
+                  DropdownMenuItem(
+                    value: 'usdt',
+                    child: Text('USDT'),
+                  ),
+                  DropdownMenuItem(
+                    value: 'volt',
+                    child: Text('VOLT'),
+                  ),
+                ],
+                decoration: InputDecoration(labelText: 'Token'),
+              ),
+              SizedBox(height: 20),
+              TextFormField(
+                controller: _amountController,
```

Comment:
Hello @NikolayRn Thank you for the feedback. The Controller is parsed in the `TextFormField`